dev/build.js

```diff
@@ -5,14 +5,20 @@ import * as esbuild from 'esbuild'
 
 import { rootDir, DEV, time } from './utils.js'
 
-try {
-  const head = await readFile(join(rootDir, '.git/HEAD'), 'utf8')
+const getHash = async head => {
+  if (!head.startsWith('ref:')) return { hash: head.trim(), branch: 'detached' }
   const parts = head.split(' ')[1].trim().split('/')
-  const last = parts[parts.length - 1]
+  const branch = parts[parts.length - 1]
   const hash = await readFile(join(rootDir, '.git', ...parts), 'utf8')
-  process.env.HASH = `${last}@${hash.trim()}`
+  return { hash: hash.trim(), branch }
+}
+  
+try {
+  const head = await readFile(join(rootDir, '.git/HEAD'), 'utf8')
+  const { hash, branch } = await getHash(head)
```

Comment:
getting the current git SHA (used for the version) was not working on detached head, so not working for cloudflare deploys !